Apps/Src/FaultState.c

```diff
@@ -12,13 +12,27 @@
 #include "CANbus.h"
 #include "BSP_UART.h"
 #include "config.h"
-#ifndef SIMULATION
-#include "stm32f4xx.h"
-#else
+#include "BSP_PLL.h"
+#ifdef SIMULATION
 #include "Simulator.h"
 extern uint8_t stateCount;
 #endif
 
+//Copy of delay_u function for Fault State
+static void delay_u_faultState(uint32_t micro)
+{
+  uint32_t delay = BSP_PLL_GetSystemClock() / 1000000;
+	for(volatile uint32_t i = 0; i < micro; i++)
+	{
+		for(volatile uint32_t j = 0; j < delay; j++);
+	}
+}
```

Comment:
this shouldn't be here. You should be using the BSP PLL function.

Reply:
![image](https://user-images.githubusercontent.com/50528867/226144276-83fb8aea-1092-4ec6-b807-618284ed18d0.png)
I'm not sure how to use this to set a delay

Reply:
You need to rebase with master. Raymond had a PR that added delay functions in the BSP_PLL library.

Reply:
If you look in master you will find them